src/classes/ALLO_Allocations_TEST.cls

```diff
@@ -117,6 +117,77 @@ private with sharing class ALLO_Allocations_TEST {
         system.assertEquals(8, queryOpps[0].Amount, 'The opportunity amount should remain unchanged after the exception is caught.');
     }
 
+    /*******************************************************************************************************
+    * @description When Payments are created after the Opportunity and Opportunity Allocations exist,
+    * ensure that the payment allocation percentages match those from the Opportunity Allocations, and
+    * are not overwritten by additional logic in the trigger.
+    * Also check that allocation rounding works correctly and no amounts are allocated to the Default GAU
+    ********************************************************************************************************/
+    static testMethod void pmtAllocationPercentCheck() {
+        General_Accounting_Unit__c defaultGau = new General_Accounting_Unit__c(Name='default GAU');
+        insert defaultGau;
+
+        setupSettings(new Allocations_Settings__c(
+            Payment_Allocations_Enabled__c = true,
+            Default_Allocations_Enabled__c = true,
+            Default__c = defaultGau.Id));
+
+        Account acc = new Account(Name='Account-pmtAlloPercentCheck');
+        insert acc;
+        Opportunity opp = new Opportunity(Name='Opp-pmtAlloPercentCheck', Amount = 100, AccountID=acc.id, CloseDate=system.today(), StageName=UTIL_UnitTestData_TEST.getClosedWonStage());
+        insert opp;
+
+        List<General_Accounting_Unit__c> gauForInsert = new List<General_Accounting_Unit__c>();
+        
+        General_Accounting_Unit__c gau1 = new General_Accounting_Unit__c(Name='GAU-1');
+        gauForInsert.add(gau1);
+        General_Accounting_Unit__c gau2 = new General_Accounting_Unit__c(Name='GAU-2');
+        gauForInsert.add(gau2);
+        General_Accounting_Unit__c gau3 = new General_Accounting_Unit__c(Name='GAU-3');
+        gauForInsert.add(gau3);
+        insert gauForInsert;
+
+        List<Allocation__c> alloForInsert = new List<Allocation__c>();
+
+        Allocation__c percentAllo1 = new Allocation__c(Opportunity__c=opp.id, Percent__c = 10, General_Accounting_Unit__c = gau1.id);
```

Comment:
Since you call this 4 times, I recommend abstracting out into a helper method. I suspect there's other code like this in this class too.

```
private static Allocation__c createAllocation(Id gauId, Id oppId, Double amt, Double percent) {
    return new Allocation__c(
            Opportunity__c = oppId, 
            General_Accounting_Unit__c = gauId,
            Percent__c = percent,
            Amount__c = amt
        );
}
```

The the lines in this method can just be:
```
         List<Allocation__c> alloForInsert = new List<Allocation__c>{
             createAllocation(gau1.Id, opp.Id, null, 10),
             createAllocation(gau1.Id, opp.Id, null, 24),
             createAllocation(gau2.Id, opp.Id, null, 33),
             createAllocation(gau1.Id, opp.Id, 33, null)
         };

Reply:
Totally agree here, I was just trying to make as few changes as possible! I'll update the rest of the class to use this format, it's much cleaner.

Reply:
Thanks @lparrott. It's ok to refactor when it helps with readability, such as to fix gross-style errors like missing curly brackets, missing spaces, or lowercase class/field names. A bonus here is a reduction in the number of lines of code in the test class.